### PR TITLE
Use messageWrapper while storing data to database

### LINKED ISSUE #2858: Filtering by message in message store throws ClassCastException
**❗ please do not add sensitive information in issues, you can provide extra information via email using issue number as reference ❗**

**Describe the issue**
Filtering messages by mesage in message store throws ClassCastException in case the listener is JavaListener.

**Reporter**
Ali

**To Reproduce**
1. Run iaf-test
2. Go to message store of a receiver that has JavaListener
3. Try to filter by message

[st.txt](https://github.com/ibissource/iaf/files/8243159/st.txt)

### PR DESCRIPTION
**Pull request by alisihab**
Closes #2858

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java [hunk 1]
```diff
@@ -630,7 +631,11 @@ private String getRawMessage(IMessageBrowser<?> messageBrowser, IListener listen
 			} else {
 				try {
 					if (listener!=null) {
-						msg = listener.extractMessage(rawmsg, null).asString();
+						if(listener instanceof JavaListener) { // TODO: remove this when JavaListener uses Message
+							msg = listener.extractMessage(Message.asString(rawmsg), null).asString();
+						} else {
+							msg = listener.extractMessage(rawmsg, null).asString();
+						}
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java:638]: This change should not be necessary, as messageBrowser returns either MessageWrapper, or M of Listener<M>

File: core/src/main/java/nl/nn/adapterframework/util/MessageBrowsingFilter.java [hunk 2]
```diff
@@ -149,15 +149,15 @@ public void setCommentMask(String commentMask) {
...
+			String msg = getRawMessageAsText(storage, listener, iterItem.getId());
 			if (!StringUtils.containsIgnoreCase(msg, message)) {
 				return false;
 			}
 		}
 		return true;
 	}
 
-	private String getRawMessage(IMessageBrowser<?> messageBrowser, IListener listener, String messageId) throws IOException, ListenerException {
+	private String getRawMessageAsText(IMessageBrowser<?> messageBrowser, IListener listener, String messageId) throws IOException, ListenerException {
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/util/MessageBrowsingFilter.java:160]: I would prefer to call this method `getMessageText()`

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java [hunk 3]
```diff
@@ -627,11 +627,17 @@ private String getRawMessage(IMessageBrowser<?> messageBrowser, IListener listen
 				} catch (IOException e) {
 					throw new ApiException(e, 500);
 				}
+			} else if(rawmsg instanceof Message){
+				try {
+					msg = ((Message)rawmsg).asString();
+				} catch (IOException e) {
+					throw new ApiException(e, 500);
+				}
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java:635]: Please add a comment that this code is only here to accommodate for a previous version of the code

**Review by gvanbrakel — APPROVED** [review]

File: core/src/main/java/nl/nn/adapterframework/util/MessageBrowsingFilter.java [hunk 4]
```diff
@@ -169,6 +169,12 @@ private String getRawMessage(IMessageBrowser<?> messageBrowser, IListener listen
 				} catch (IOException e) {
 					throw new ApiException(e, 500);
 				}
+			} else if(rawmsg instanceof Message) { // For backwards compatibility: in case the message is stored as Message and the listener uses a different type.
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/util/MessageBrowsingFilter.java:172]:
```suggestion
			} else if(rawmsg instanceof Message) { // For backwards compatibility: earlier MessageLog messages were stored as Message.
```

File: core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java [hunk 5]
```diff
@@ -627,11 +627,17 @@ private String getRawMessage(IMessageBrowser<?> messageBrowser, IListener listen
 				} catch (IOException e) {
 					throw new ApiException(e, 500);
 				}
+			} else if(rawmsg instanceof Message) { // For backwards compatibility: in case the message is stored as Message and the listener uses a different type.
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java:630]:
```suggestion
			} else if(rawmsg instanceof Message) { // For backwards compatibility: earlier MessageLog messages were stored as Message.
```

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**